# 1) EXPLORATORY DATA ANALYSIS.

In [36]:
import pandas as pd
import numpy as np

#Read files:
train = pd.read_csv("E:/Downloads/Train_bm.csv")
test = pd.read_csv("E:/Downloads/Test_bm.csv")

Its generally a good idea to combine both train and test data sets into one, perform feature engineering and then divide them later again. This saves the trouble of performing the same steps twice on test and train. Lets combine them into a dataframe ‘data’ with a ‘source’ column specifying where each observation belongs.

In [37]:
train['source']='train' 
test['source']='test'
data = pd.concat([train, test],ignore_index=True)
print train.shape, test.shape, data.shape

(8523, 13) (5681, 12) (14204, 13)


Thus we can see that data has same columns but rows equivalent to both test and train. One of the key challenges in any data set is missing values. Lets start by checking which columns contain missing values.

In [38]:
data.apply(lambda x: sum(x.isnull()))

Item_Fat_Content                0
Item_Identifier                 0
Item_MRP                        0
Item_Outlet_Sales            5681
Item_Type                       0
Item_Visibility                 0
Item_Weight                  2439
Outlet_Establishment_Year       0
Outlet_Identifier               0
Outlet_Location_Type            0
Outlet_Size                  4016
Outlet_Type                     0
source                          0
dtype: int64

In [39]:
data

Item_Fat_Content Item_Identifier  Item_MRP  Item_Outlet_Sales  \
0              Low Fat           FDA15  249.8092          3735.1380   
1              Regular           DRC01   48.2692           443.4228   
2              Low Fat           FDN15  141.6180          2097.2700   
3              Regular           FDX07  182.0950           732.3800   
4              Low Fat           NCD19   53.8614           994.7052   
5              Regular           FDP36   51.4008           556.6088   
6              Regular           FDO10   57.6588           343.5528   
7              Low Fat           FDP10  107.7622          4022.7636   
8              Regular           FDH17   96.9726          1076.5986   
9              Regular           FDU28  187.8214          4710.5350   
10             Low Fat           FDY07   45.5402          1516.0266   
11             Regular           FDA03  144.1102          2187.1530   
12             Regular           FDX32  145.4786          1589.2646   
13             Regular           FDS46  119.6782          2145.2076   
14             Low Fat           FDF32  196.4426          1977.4260   
15             Regular           FDP49   56.3614          1547.3192   
16             Low Fat           NCB42  115.3492          1621.8888   
17             Regular           FDP49   54.3614           718.3982   
18             Low Fat           DRI11  113.2834          2303.6680   
19             Low Fat           FDU02  230.5352          2748.4224   
20             Regular           FDN22  250.8724          3775.0860   
21             Regular           FDW12  144.5444          4064.0432   
22             Low Fat           NCB30  196.5084          1587.2672   
23             Low Fat           FDC37  107.6938           214.3876   
24             Regular           FDR28  165.0210          4078.0250   
25             Low Fat           NCD06   45.9060           838.9080   
26             Regular           FDV10   42.3112          1065.2800   
27             low fat           DRJ59   39.1164           308.9312   
28             Regular           FDE51   45.5086           178.4344   
29             Regular           FDC14   43.6454           125.8362   
...                ...             ...       ...                ...   
14174          Regular           FDD23  186.1898                NaN   
14175          Low Fat           FDP32  126.7678                NaN   
14176          Regular           FDO31   80.2960                NaN   
14177          Low Fat           FDQ57  144.5760                NaN   
14178          Regular           FDX32  143.6786                NaN   
14179          Low Fat           FDC39  207.8296                NaN   
14180          Low Fat           FDU55  261.3278                NaN   
14181          Low Fat           FDL45  123.4704                NaN   
14182          Low Fat           DRK37  188.8530                NaN   
14183          Low Fat           FDK22  215.3850                NaN   
14184          Low Fat           DRG37  155.7972                NaN   
14185          Low Fat           FDK22  214.3850                NaN   
14186          Low Fat           DRH36   72.8696                NaN   
14187          Regular           DRC36  173.4054                NaN   
14188          Low Fat           DRE03   46.0718                NaN   
14189          Regular           FDF34  196.9084                NaN   
14190          Low Fat           FDZ22   82.1250                NaN   
14191          Low Fat           FDC44  115.1518                NaN   
14192          Low Fat           FDN31  188.0530                NaN   
14193          Regular           FDO03  229.4352                NaN   
14194              reg           FDA01   59.5904                NaN   
14195          Low Fat           NCH42  231.1010                NaN   
14196          Low Fat           FDF46  116.0834                NaN   
14197          Low Fat           DRL35   43.2770                NaN   
14198          Regular        

Lets look at some basic statistics for numerical variables.



In [40]:
data.describe()

Item_MRP  Item_Outlet_Sales  Item_Visibility   Item_Weight  \
count  14204.000000        8523.000000     14204.000000  11765.000000   
mean     141.004977        2181.288914         0.065953     12.792854   
std       62.086938        1706.499616         0.051459      4.652502   
min       31.290000          33.290000         0.000000      4.555000   
25%       94.012000         834.247400         0.027036      8.710000   
50%      142.247000        1794.331000         0.054021     12.600000   
75%      185.855600        3101.296400         0.094037     16.750000   
max      266.888400       13086.964800         0.328391     21.350000   

       Outlet_Establishment_Year  
count               14204.000000  
mean                 1997.830681  
std                     8.371664  
min                  1985.000000  
25%                  1987.000000  
50%                  1999.000000  
75%                  2004.000000  
max                  2009.000000

Moving to nominal (categorical) variable, lets have a look at the number of unique values in each of them.

In [41]:
data.apply(lambda x: len(x.unique()))

Item_Fat_Content                 5
Item_Identifier               1559
Item_MRP                      8052
Item_Outlet_Sales             3494
Item_Type                       16
Item_Visibility              13006
Item_Weight                    416
Outlet_Establishment_Year        9
Outlet_Identifier               10
Outlet_Location_Type             3
Outlet_Size                      4
Outlet_Type                      4
source                           2
dtype: int64

# This tells us that there are 1559 products and 10 outlets/stores.

# Let’s explore further using the frequency of different categories in each nominal variable. I’ll exclude the ID and source variables for obvious reasons.

In [42]:
data.dtypes

Item_Fat_Content              object
Item_Identifier               object
Item_MRP                     float64
Item_Outlet_Sales            float64
Item_Type                     object
Item_Visibility              float64
Item_Weight                  float64
Outlet_Establishment_Year      int64
Outlet_Identifier             object
Outlet_Location_Type          object
Outlet_Size                   object
Outlet_Type                   object
source                        object
dtype: object

In [43]:
data.dtypes.index

Index([u'Item_Fat_Content', u'Item_Identifier', u'Item_MRP',
       u'Item_Outlet_Sales', u'Item_Type', u'Item_Visibility', u'Item_Weight',
       u'Outlet_Establishment_Year', u'Outlet_Identifier',
       u'Outlet_Location_Type', u'Outlet_Size', u'Outlet_Type', u'source'],
      dtype='object')

# Data types/ str


Retrieve the object data types from the above and perform value counts for knowing the frequency.

# Frequencies and spell check.

In [44]:
#Filter categorical variables
categorical_columns = [x for x in data.dtypes.index if data.dtypes[x]=='object']
#Exclude ID cols and source:
categorical_columns = [x for x in categorical_columns if x not in ['Item_Identifier','Outlet_Identifier','source']]
#Print frequency of categories
for col in categorical_columns:
    print '\nFrequency of Categories for varible %s'%col
    print data[col].value_counts()


Frequency of Categories for varible Item_Fat_Content
Low Fat    8485
Regular    4824
LF          522
reg         195
low fat     178
Name: Item_Fat_Content, dtype: int64

Frequency of Categories for varible Item_Type
Fruits and Vegetables    2013
Snack Foods              1989
Household                1548
Frozen Foods             1426
Dairy                    1136
Baking Goods             1086
Canned                   1084
Health and Hygiene        858
Meat                      736
Soft Drinks               726
Breads                    416
Hard Drinks               362
Others                    280
Starchy Foods             269
Breakfast                 186
Seafood                    89
Name: Item_Type, dtype: int64

Frequency of Categories for varible Outlet_Location_Type
Tier 3    5583
Tier 2    4641
Tier 1    3980
Name: Outlet_Location_Type, dtype: int64

Frequency of Categories for varible Outlet_Size
Medium    4655
Small     3980
High      1553
Name: Outlet_Size, dtype: int64

F

# The above step is used to know whether the data has any wrongly spelled words or not 

#### ex: Low Fat , LF , low fat ,etc....

The output gives us following observations:

Item_Fat_Content: Some of ‘Low Fat’ values mis-coded as ‘low fat’ and ‘LF’. Also, some of ‘Regular’ are mentioned as ‘regular’.
Item_Type: Not all categories have substantial numbers. It looks like combining them can give better results.
Outlet_Type: Supermarket Type2 and Type3 can be combined. But we should check if that’s a good idea before doing it.

In [45]:
item_avg_weight = data.pivot_table(values='Item_Weight', index='Item_Identifier')

In [46]:
item_avg_weight

Item_Weight
Item_Identifier             
DRA12                 11.600
DRA24                 19.350
DRA59                  8.270
DRB01                  7.390
DRB13                  6.115
DRB24                  8.785
DRB25                 12.300
DRB48                 16.750
DRC01                  5.920
DRC12                 17.850
DRC13                  8.260
DRC24                 17.850
DRC25                  5.730
DRC27                 13.800
DRC36                 13.000
DRC49                  8.670
DRD01                 12.100
DRD12                  6.960
DRD13                 15.000
DRD15                 10.600
DRD24                 13.850
DRD25                  6.135
DRD27                 18.750
DRD37                  9.800
DRD49                  9.895
DRD60                 15.700
DRE01                 10.100
DRE03                 19.600
DRE12                  4.590
DRE13                  6.280
...                      ...
NCX05                 15.200
NCX06                 17.600
NCX17                 21.250
NCX18                 14.150
NCX29                 10.000
NCX30                 16.700
NCX41                 19.000
NCX42                  6.360
NCX53                 20.100
NCX54                  9.195
NCY05                 13.500
NCY06                 15.250
NCY17                 18.200
NCY18                  7.285
NCY29                 13.650
NCY30                 20.250
NCY41                 16.750
NCY42                  6.380
NCY53                 20.000
NCY54                  8.430
NCZ05                  8.485
NCZ06                 19.600
NCZ17                 12.150
NCZ18                  7.825
NCZ29                 15.000
NCZ30                  6.590
NCZ41                 19.850
NCZ42                 10.500
NCZ53                  9.600
NCZ54                 14.650

[1559 rows x 1 columns]

#### The above is a pivot table

# 2) DATA CLEANSING.

In [47]:
data.isnull().sum() 

Item_Fat_Content                0
Item_Identifier                 0
Item_MRP                        0
Item_Outlet_Sales            5681
Item_Type                       0
Item_Visibility                 0
Item_Weight                  2439
Outlet_Establishment_Year       0
Outlet_Identifier               0
Outlet_Location_Type            0
Outlet_Size                  4016
Outlet_Type                     0
source                          0
dtype: int64

In [48]:
data.fillna(data.mean(), inplace=True) # Only applicable when we don't have binary data like 0's and 1's.When 0 or 1,
# we need to go by mode of 0's and 1's.

In [49]:
data.isnull().sum()

Item_Fat_Content                0
Item_Identifier                 0
Item_MRP                        0
Item_Outlet_Sales               0
Item_Type                       0
Item_Visibility                 0
Item_Weight                     0
Outlet_Establishment_Year       0
Outlet_Identifier               0
Outlet_Location_Type            0
Outlet_Size                  4016
Outlet_Type                     0
source                          0
dtype: int64

# Above is for numerical missing values & below is for categorical (or) binary missing valies

In [50]:
data['Outlet_Size'].mode()

0    Medium
dtype: object

In [51]:
#Data.COLUMN_NAME.fillna(0,inplace=True)
data.Outlet_Size.fillna('Medium',inplace=True)

In [52]:
data.isna().sum()

Item_Fat_Content             0
Item_Identifier              0
Item_MRP                     0
Item_Outlet_Sales            0
Item_Type                    0
Item_Visibility              0
Item_Weight                  0
Outlet_Establishment_Year    0
Outlet_Identifier            0
Outlet_Location_Type         0
Outlet_Size                  0
Outlet_Type                  0
source                       0
dtype: int64

# 3) FEATURE ENGINEERING.

In [53]:
# Checking if there any combining of type-2 and type-3 supermarket since they have low frequencies.
data.pivot_table(values='Item_Outlet_Sales',index='Outlet_Type')


Item_Outlet_Sales
Outlet_Type                         
Grocery Store            1076.412666
Supermarket Type1        2262.232963
Supermarket Type2        2069.766739
Supermarket Type3        3088.550567

## Not ideal to combine both type-2 & type-3 as both have significant difference with output.

### Pivot table is used to check the categorical values with any continuos variable as above.

# #Impute 0 values with mean visibility of that product:
     AS THE VISIBILITY OF THE PRODUCT WILL NEVER BE ZERO

# ################################################BLank###############################################################

# Rename them to more intuitive categories:


In [64]:
#Get the first two characters of ID:
data['Item_Type_Combined'] = data['Item_Identifier'].apply(lambda x: x[0:2])


In [65]:
data.head()

Item_Fat_Content Item_Identifier  Item_MRP  Item_Outlet_Sales  \
0          Low Fat           FDA15  249.8092          3735.1380   
1          Regular           DRC01   48.2692           443.4228   
2          Low Fat           FDN15  141.6180          2097.2700   
3          Regular           FDX07  182.0950           732.3800   
4          Low Fat           NCD19   53.8614           994.7052   

               Item_Type                                    Item_Visibility  \
0                  Dairy                                          0.0160473   
1            Soft Drinks                                          0.0192782   
2                   Meat                                          0.0167601   
3  Fruits and Vegetables                   Item_Visibility
Item_Identifi...   
4              Household                   Item_Visibility
Item_Identifi...   

   Item_Weight  Outlet_Establishment_Year Outlet_Identifier  \
0         9.30                       1999            OUT049   
1         5.92                       2009            OUT018   
2        17.50                       1999            OUT049   
3        19.20                       1998            OUT010   
4         8.93                       1987            OUT013   

  Outlet_Location_Type Outlet_Size        Outlet_Type source  \
0               Tier 1      Medium  Supermarket Type1  train   
1               Tier 3      Medium  Supermarket Type2  train   
2               Tier 1      Medium  Supermarket Type1  train   
3               Tier 3      Medium      Grocery Store  train   
4               Tier 3        High  Supermarket Type1  train   

  Item_Type_Combined  
0                 FD  
1                 DR  
2                 FD  
3                 FD  
4                 NC

In [66]:
data['Item_Type_Combined'] = data['Item_Type_Combined'].map({'FD':'Food',
                                                             'NC':'Non-Consumable',
                                                             'DR':'Drinks'})

In [67]:
data.head()

Item_Fat_Content Item_Identifier  Item_MRP  Item_Outlet_Sales  \
0          Low Fat           FDA15  249.8092          3735.1380   
1          Regular           DRC01   48.2692           443.4228   
2          Low Fat           FDN15  141.6180          2097.2700   
3          Regular           FDX07  182.0950           732.3800   
4          Low Fat           NCD19   53.8614           994.7052   

               Item_Type                                    Item_Visibility  \
0                  Dairy                                          0.0160473   
1            Soft Drinks                                          0.0192782   
2                   Meat                                          0.0167601   
3  Fruits and Vegetables                   Item_Visibility
Item_Identifi...   
4              Household                   Item_Visibility
Item_Identifi...   

   Item_Weight  Outlet_Establishment_Year Outlet_Identifier  \
0         9.30                       1999            OUT049   
1         5.92                       2009            OUT018   
2        17.50                       1999            OUT049   
3        19.20                       1998            OUT010   
4         8.93                       1987            OUT013   

  Outlet_Location_Type Outlet_Size        Outlet_Type source  \
0               Tier 1      Medium  Supermarket Type1  train   
1               Tier 3      Medium  Supermarket Type2  train   
2               Tier 1      Medium  Supermarket Type1  train   
3               Tier 3      Medium      Grocery Store  train   
4               Tier 3        High  Supermarket Type1  train   

  Item_Type_Combined  
0               Food  
1             Drinks  
2               Food  
3               Food  
4     Non-Consumable

In [68]:
data['Item_Type_Combined'].value_counts()

Food              10201
Non-Consumable     2686
Drinks             1317
Name: Item_Type_Combined, dtype: int64

# Years:


In [69]:
data['Outlet_Years'] = 2013 - data['Outlet_Establishment_Year']
data['Outlet_Years'].describe()

count    14204.000000
mean        15.169319
std          8.371664
min          4.000000
25%          9.000000
50%         14.000000
75%         26.000000
max         28.000000
Name: Outlet_Years, dtype: float64

In [70]:
data.head()

Item_Fat_Content Item_Identifier  Item_MRP  Item_Outlet_Sales  \
0          Low Fat           FDA15  249.8092          3735.1380   
1          Regular           DRC01   48.2692           443.4228   
2          Low Fat           FDN15  141.6180          2097.2700   
3          Regular           FDX07  182.0950           732.3800   
4          Low Fat           NCD19   53.8614           994.7052   

               Item_Type                                    Item_Visibility  \
0                  Dairy                                          0.0160473   
1            Soft Drinks                                          0.0192782   
2                   Meat                                          0.0167601   
3  Fruits and Vegetables                   Item_Visibility
Item_Identifi...   
4              Household                   Item_Visibility
Item_Identifi...   

   Item_Weight  Outlet_Establishment_Year Outlet_Identifier  \
0         9.30                       1999            OUT049   
1         5.92                       2009            OUT018   
2        17.50                       1999            OUT049   
3        19.20                       1998            OUT010   
4         8.93                       1987            OUT013   

  Outlet_Location_Type Outlet_Size        Outlet_Type source  \
0               Tier 1      Medium  Supermarket Type1  train   
1               Tier 3      Medium  Supermarket Type2  train   
2               Tier 1      Medium  Supermarket Type1  train   
3               Tier 3      Medium      Grocery Store  train   
4               Tier 3        High  Supermarket Type1  train   

  Item_Type_Combined  Outlet_Years  
0               Food            14  
1             Drinks             4  
2               Food            14  
3               Food            15  
4     Non-Consumable            26

# Change categories of low fat:


In [71]:

print 'Original Categories:'
print data['Item_Fat_Content'].value_counts()

print '\nModified Categories:'
data['Item_Fat_Content'] = data['Item_Fat_Content'].replace({'LF':'Low Fat',
                                                             'reg':'Regular',
                                                             'low fat':'Low Fat'})
print data['Item_Fat_Content'].value_counts()

Original Categories:
Low Fat    8485
Regular    4824
LF          522
reg         195
low fat     178
Name: Item_Fat_Content, dtype: int64

Modified Categories:
Low Fat    9185
Regular    5019
Name: Item_Fat_Content, dtype: int64


In [72]:
#Mark non-consumables as separate category in low_fat:
data.loc[data['Item_Type_Combined']=="Non-Consumable",'Item_Fat_Content'] = "Non-Edible"
data['Item_Fat_Content'].value_counts()

Low Fat       6499
Regular       5019
Non-Edible    2686
Name: Item_Fat_Content, dtype: int64

# Dummies

In [76]:
data.head()

Item_Fat_Content Item_Identifier  Item_MRP  Item_Outlet_Sales  \
0          Low Fat           FDA15  249.8092          3735.1380   
1          Regular           DRC01   48.2692           443.4228   
2          Low Fat           FDN15  141.6180          2097.2700   
3          Regular           FDX07  182.0950           732.3800   
4       Non-Edible           NCD19   53.8614           994.7052   

               Item_Type                                    Item_Visibility  \
0                  Dairy                                          0.0160473   
1            Soft Drinks                                          0.0192782   
2                   Meat                                          0.0167601   
3  Fruits and Vegetables                   Item_Visibility
Item_Identifi...   
4              Household                   Item_Visibility
Item_Identifi...   

   Item_Weight  Outlet_Establishment_Year Outlet_Identifier  \
0         9.30                       1999            OUT049   
1         5.92                       2009            OUT018   
2        17.50                       1999            OUT049   
3        19.20                       1998            OUT010   
4         8.93                       1987            OUT013   

  Outlet_Location_Type Outlet_Size        Outlet_Type source  \
0               Tier 1      Medium  Supermarket Type1  train   
1               Tier 3      Medium  Supermarket Type2  train   
2               Tier 1      Medium  Supermarket Type1  train   
3               Tier 3      Medium      Grocery Store  train   
4               Tier 3        High  Supermarket Type1  train   

  Item_Type_Combined  Outlet_Years  
0               Food            14  
1             Drinks             4  
2               Food            14  
3               Food            15  
4     Non-Consumable            26

In [77]:
data=pd.get_dummies(data,columns=['Item_Fat_Content','Outlet_Location_Type','Outlet_Size','Outlet_Type','Item_Type_Combined'])

In [78]:
data.head()

Item_Identifier  Item_MRP  Item_Outlet_Sales              Item_Type  \
0           FDA15  249.8092          3735.1380                  Dairy   
1           DRC01   48.2692           443.4228            Soft Drinks   
2           FDN15  141.6180          2097.2700                   Meat   
3           FDX07  182.0950           732.3800  Fruits and Vegetables   
4           NCD19   53.8614           994.7052              Household   

                                     Item_Visibility  Item_Weight  \
0                                          0.0160473         9.30   
1                                          0.0192782         5.92   
2                                          0.0167601        17.50   
3                   Item_Visibility
Item_Identifi...        19.20   
4                   Item_Visibility
Item_Identifi...         8.93   

   Outlet_Establishment_Year Outlet_Identifier source  Outlet_Years  \
0                       1999            OUT049  train            14   
1                       2009            OUT018  train             4   
2                       1999            OUT049  train            14   
3                       1998            OUT010  train            15   
4                       1987            OUT013  train            26   

                 ...                  Outlet_Size_High  Outlet_Size_Medium  \
0                ...                                 0                   1   
1                ...                                 0                   1   
2                ...                                 0                   1   
3                ...                                 0                   1   
4                ...                                 1                   0   

   Outlet_Size_Small  Outlet_Type_Grocery Store  \
0                  0                          0   
1                  0                          0   
2                  0                          0   
3                  0                          1   
4                  0                          0   

   Outlet_Type_Supermarket Type1  Outlet_Type_Supermarket Type2  \
0                              1                              0   
1                              0                              1   
2                              1                              0   
3                              0                              0   
4                              1                              0   

   Outlet_Type_Supermarket Type3  Item_Type_Combined_Drinks  \
0                              0                          0   
1                              0                          1   
2                              0                          0   
3                              0                          0   
4                              0                          0   

   Item_Type_Combined_Food  Item_Type_Combined_Non-Consumable  
0                        1                                  0  
1                        0                                  0  
2                        1                                  0  
3                        1                                  0  
4                        0                                  1  

[5 rows x 26 columns]

In [79]:
data.dtypes


Item_Identifier                       object
Item_MRP                             float64
Item_Outlet_Sales                    float64
Item_Type                             object
Item_Visibility                       object
Item_Weight                          float64
Outlet_Establishment_Year              int64
Outlet_Identifier                     object
source                                object
Outlet_Years                           int64
Item_Fat_Content_Low Fat               uint8
Item_Fat_Content_Non-Edible            uint8
Item_Fat_Content_Regular               uint8
Outlet_Location_Type_Tier 1            uint8
Outlet_Location_Type_Tier 2            uint8
Outlet_Location_Type_Tier 3            uint8
Outlet_Size_High                       uint8
Outlet_Size_Medium                     uint8
Outlet_Size_Small                      uint8
Outlet_Type_Grocery Store              uint8
Outlet_Type_Supermarket Type1          uint8
Outlet_Type_Supermarket Type2          uint8
Outlet_Typ

In [81]:
data.drop(['Item_Type','Outlet_Establishment_Year'],axis=1,inplace=True)


# Divide into test and train:


In [82]:
train = data.loc[data['source']=="train"]
test = data.loc[data['source']=="test"]


# Dropping columns

In [84]:
test.drop(['Item_Outlet_Sales','source'],axis=1,inplace=True)
train.drop(['source'],axis=1,inplace=True)

C:\Users\lenovo\Anaconda2\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.
C:\Users\lenovo\Anaconda2\lib\site-packages\ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


# Writing to new files all together

In [86]:
train.to_csv("train_modified.csv",index=False)
test.to_csv("test_modified.csv",index=False)

# #NOW WE HAVE A NEW TRAINING AND TEST SETS. THEREFORE WE WILL RUN OUR TRADITIONAL ALGORITHMS REQUIRED FOR THIS LATER ON.